jplag/src/main/java/de/jplag/Submission.java

```diff
@@ -147,10 +149,10 @@ public int getSimilarityDivisor(boolean subtractBaseCode) {
     }
 
     /**
-     * @return Parse result of the submission.
+     * @return unmodifiable list of tokens generated by parsing the submission.
      */
-    public TokenList getTokenList() {
-        return tokenList;
+    public List<Token> getTokenList() {
+        return tokenList == null ? null : Collections.unmodifiableList(tokenList);
```

Comment:
As already mentioned, I am not a fan of an unmodifiable list. I would prefer to return a read-only view.

Reply:
Agreed in maintainer meeting to use unmodifiable lists with an according documentation. Thus this resolves without further action.